### ISSUE TITLE
Can Swagger generate custom generic types?

### ISSUE THREAD

**Issue by MelnikovIG** [swagger-api/swagger-codegen#8194]
Assume we have API return model in C#

    public class ApiResult<T>
    {
      public T Result;
      public bool Success;
    }

and returning `ApiResult<string>` object instance to client

so we have swagger generated model

    ApiResult[String] {
      result (string, optional),
      success (boolean, optional)
    }

which is incorrectly converted to typescript class using https://swagger.io/swagger-codegen/

    'use strict';
    import * as models from './models';
    export interface ApiResultString {
        result?: string;
        success?: boolean;
    }

Is it possible to generate output models with generics same as in input models?

**Comment by Pangamma** [comment]
Bump

**Comment by HugoMario** [comment]
i'm not so familiar with `typescript`, but if you guys conclude that this is the best practice, we can implement it.

**Comment by demonhell** [comment]
I agree with the opening post. 
Generally `ApiResult` is a generic wrapper to the real api answer from the rest backend.

At this moment if you have n APIs that all wraps n different objects, you end having `ApiResultObjectApi1... ApiResultObjectApi2... ApiResultObjectApiN` in the generated code.
Several useless repetition that serves a "difficult" to use `api.service.ts`

Since typescript supports generics, it would be ideal to generate the typescript interfaces as generics as well. The code that use those could use more generic  and the models would be DRY and smaller.

**Comment by demonhell** [comment]
A little more thinking into this and i just  realized that there is an indecision case.

Suppose a c# mode like this
```
public class ApiResult<T>
{
  public T Result;
  public bool Success;
  public string message;
}
```
and an api returning `ApiResult<string>` object instance to client

so we would have swagger generated model

```
ApiResult[String] {
  result (string, optional),
  success (boolean, optional)
  message (string, optional)
}
```

how the generator's template should behave? 
It doesn't know which of the strings are from the generics.

Maybe a typescript-angular generate's option like `enableGenerics` (default: false) should be better. so you can enable it only iff you're sure of the api's model.

**Comment by Pangamma** [comment]
Yes, the spec needs to be updated to handle this case. Otherwise, a generator would need a minimum of two types being used in a generic result before it could decide which properties are generic and which properties are stationary.

**Comment by robbertvanhove** [comment]
Hi, is there any progress on supporting generics in TypeScript yet? Thanx!

**Comment by bluoxy** [comment]
I have a same problem.

**Comment by Savrov** [comment]
I wish to use this feature too

**Comment by fendorio** [comment]
Same

**Comment by Bernix01** [comment]
Any update on this?

**Comment by Pangamma** [comment]
2021 says hello. I still need and want this feature.

**Comment by rhopiCoder** [comment]
+1

**Comment by Thomas-Lederer** [comment]
+1

**Comment by damien7792** [comment]
+1

**Comment by ShuuMakengi** [comment]
+1

**Comment by joshuafc** [comment]
+1

**Comment by fadi-shahoud** [comment]
+1

**Comment by barta-home** [comment]
Same here

**Comment by joshuafc** [comment]
2022 says hello. I still need and want this feature.

**Comment by duaneking** [comment]
Bumping for fix

**Comment by m17kea** [comment]
Would love to see this implemented

**Comment by josemariabaena** [comment]
i have the same problem . is is available?

**Comment by ddallago-betclic** [comment]
+1

**Comment by davethieben** [comment]
2023 says, siighhh.... still open...

**Comment by Chgocn** [comment]
I wish to use this feature too

**Comment by ullfindsmit** [comment]
+1

**Comment by GaffaRay** [comment]
Great feature to implement.

**Comment by sabine19** [comment]
+1

**Comment by Junad-Park** [comment]
still open...

**Comment by Aweptimum** [comment]
It's possible through dynamic references ([henry andrews approved](https://github.com/OAI/OpenAPI-Specification/issues/957#issuecomment-1912910504)), which I don't think swagger-codegen supports

**Comment by Alamonall** [comment]
Hello! Any progress?
